src/main/java/de/tum/in/www1/artemis/service/compass/CompassService.java

```diff
@@ -53,6 +54,13 @@
 
     private final ConflictingResultService conflictingResultService;
 
+    /**
+     * Map submissionIds to automatic results. Automatic results generated by Compass are not stored in the database, instead they are stored in this map. As soon as a submission
+     * is locked for assessment, the corresponding automatic result will be retrieved from this map and stored to the database.
+     */
+    // TODO CZ: this map contains the automatic results for the submissions of all exercises. Should we create a separate automatic result map for the different exercises?
```

Comment:
Please have a look how we handle this in quizzes. I probably makes sense to split this into multiple ConcurrentHashMaps per exercise, because this could decrease the changes of problems due to concurrent access

Reply:
I changed the `Map<Long, Result>` to a `Map<Long, Map<Long, Result>>` that separates the automatic results by exercuse,